lib/setup.c

```diff
@@ -921,6 +926,40 @@ ___SIZE_TS requested_words_still;)
 }
 
 
+___EXP_FUNC(___BOOL, ___fdset_resize)
+   ___P((int fd1,
+         int fd2),
+        (fd1,
+         fd2)
+int fd1;
+int fd2;)
+{
+#ifdef USE_select_or_poll
+
+  ___processor_state ___ps = ___PSTATE;
+  int fd = (fd2 > fd1) ? fd2 : fd1;
+
+  if (fd < ___ps->os.fdset.size)
+    return 0;
+
+  ___fdset_resize_heap_overflow_clear ();
+
+  ___sync_op_struct sop;
+
+  sop.op = OP_FDSET_RESIZE;
+  sop.arg[0] = fd;
+
+  on_all_processors (___PSP &sop);
+
+  return ___fdset_resize_heap_overflow ();
```

Comment:
I think we need a separate sync op to clear overflow instead of just a barrier.

So the logic becomes:
```
___fdset_reisze():
  on_all_processors (OP_FDSET_CLEAROVL)
  newsize = ... // move the actual sizing operation outsize ___fdset_realloc
  on_all_processors (OP_FDSET_RESIZE)
```